src/mem/cache/prefetch/indirect_memory_v2.hh

```diff
@@ -0,0 +1,662 @@
+/**
```

Comment:
I think that this copyright will not suffice. Please check LICENSE for a template of the license we use in gem5. Some companies require adding some extra information (for example, check src/arch/arm/decoder.hh), so you may want to check with the people responsible for the copyright info at Samsung.